opal/mca/common/ofi/common_ofi.h

```diff
@@ -33,39 +33,41 @@ typedef struct opal_common_ofi_module {
 
 extern opal_common_ofi_module_t opal_common_ofi;
 
-
 /**
- * Register component-specialized MCA variables
+ * Common MCA registration
  *
- * Register MCA variables common to all OFI components on behalf of
- * the calling component.  Expected to be called during
- * component_register for all OFI-related components.
+ * Common MCA registration handlinge.  After calling this function,
+ * \code opal_common_ofi.output will be properly initialized.
  *
  * @param component (IN) OFI component being initialized
  *
  * @returns OPAL_SUCCESS on success, OPAL error code on failure
  */
-OPAL_DECLSPEC int opal_common_ofi_register_mca_variables(const mca_base_component_t *component);
+OPAL_DECLSPEC int opal_common_ofi_mca_register(const mca_base_component_t *component);
 
 /**
- * Common MCA registration
+ * Initializes common objects for libfabric
  *
- * Common MCA registration handlinge.  After calling this function,
- * \code opal_common_ofi.output will be properly initialized.
+ * Initialize common libfabric interface.  This should be called from
+ * any other OFI component's component_open() call.
  *
- * @returns OPAL_SUCCESS on success, OPAL error code on failure
+ * @note This function is not thread safe and must be called in a
+ * serial portion of the code.
  */
-OPAL_DECLSPEC void opal_common_ofi_mca_register(void);
+OPAL_DECLSPEC int opal_common_ofi_open(void);
 
 /**
- * Common MCA cleanup
+ * Cleans up common objects for libfabric
  *
- * Cleanup for any resources registered during \code
- * opal_common_ofi_mca_register().
+ * Clean up common libfabric interface.  This should be called from
+ * any other OFI component's component_close() call.  Resource cleanup
+ * is reference counted, so any successful call to
+ * opal_common_ofi_init().
  *
- * @returns OPAL_SUCCESS on success, OPAL error code on failure
+ * @note This function is not thread safe and must be called in a
```

Comment:
Is component open/close of two different framework components always called serially? I threw in a lock in my patch because I wasn't sure about this and made ofi_open and ofi_close thread safe

Reply:
> Is component open/close of two different framework components always called serially? I threw in a lock in my patch because I wasn't sure about this and made ofi_open and ofi_close thread safe

Yes, always serial.  There may be other threads, but they will not be opening components.